OpenChange/MAPIStoreMailMessage.m

```diff
@@ -316,15 +316,23 @@ - (void) _fetchHeaderData
           if (i == 0)
              {
                ASSIGN (headerMimeType, mimeType);
-               ASSIGN (headerCharset, charset);               
                parameters = partParameters;
              }
-          else
+
+          if ([mimeType isEqualToString: @"text/plain"])
             {
-              /* We can only put one header charset so we will use utf-8
-                 when we have different charsets in the parts */
-              if (headerCharset != charset)
-                ASSIGN (headerCharset, @"utf-8");
+              //  utf-8 is always encoded in text/plain
+              ASSIGN (headerCharset, @"utf-8");              
```

Comment:
text/plain has been always encoded as utf8 in MAPIStoreMailMessage, so in practice it would be as if the part encoding is utf8.